tests/fuzzing/fuzz_translate_generic.c

```diff
@@ -0,0 +1,109 @@
+//
+// liblouis Braille Translation and Back-Translation Library
+//
+// Copyright (C) 2022 Anna Stan, Nicolas Morel, Kalilou Mamadou Dramé
+//
+// This file is part of liblouis.
+//
+// liblouis is free software: you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License as published
+// by the Free Software Foundation, either version 2.1 of the License, or
+// (at your option) any later version.
+//
+// liblouis is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with liblouis. If not, see <http://www.gnu.org/licenses/>.
+//
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <internal.h>
+#include <liblouis.h>
+#include <assert.h>
+
+#define LANGUAGE	"en"
+
+static int initialized = 0;
+
+#define BOLDRED(x)	"\x1b[31m\x1b[1m" x "\x1b[0m"
+
+static const char *table_default;
+
+static void __attribute__((destructor))
+free_ressources(void) {
+	lou_free();
+}
+
+void
+avoid_log(logLevels level, const char *msg) {
+	(void) level;
+	(void) msg;
+}
+
+extern int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size);
+
+int
+LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+	int inputLen = 0;
+	int outputLen = 0;
+    
+    if (size < 512) {
```

Comment:
This might seen like nitpicking, but why 512? If you want the first half of the fuzzing data why don't you just take `size/2`?

Reply:
This is because making it size/2 would mean a minor mutation in the corpus can have large impact on the way the target executes -- I think too large with respect to making it easy for the fuzzing engine to mutate a corpus and advanced coverage a bit in each iteration. The hypothesis is setting the table size fixed will improve the way the fuzzer works based on the underlying algorithms of the fuzzing engine.

Reply:
OK, but this does mean that we'll only ever have tables of length 512 bytes. 

I was under the impression that you created the generic fuzzer with the intent to allow for more axes of variability. So I thought why restrict the table size.

But sure we can go with this restriction. Maybe it really will fluctuate too much without it.

Reply:
> OK, but this does mean that we'll only ever have tables of length 512 bytes.

No, it means that it can be up to 512 bytes -- the fuzzer can always decide to add e.g. whitespace. I'm happy to leave it as is and if it proves that this is not enough to explore the code my suggestion would be to increase it.

Reply:
OK, sounds good thanks